src/mongoose_account_api.erl

```diff
@@ -0,0 +1,320 @@
+                      TimeStampNow :: non_neg_integer(),
+                      SecOlder :: non_neg_integer()) -> boolean().
+delete_old_user({LUser, LServer}, TimeStampNow, SecOlder) ->
+    %% Check if the user is logged
+    JID = jid:make(LUser, LServer, <<>>),
+    case ejabberd_sm:get_user_resources(JID) of
+        [] -> delete_old_user_if_nonactive_long_enough(JID, TimeStampNow, SecOlder);
+        _ -> false
+    end.
+
+-spec delete_old_user_if_nonactive_long_enough(JID :: jid:jid(),
+                                               TimeStampNow :: non_neg_integer(),
+                                               SecOlder :: non_neg_integer()) -> boolean().
+delete_old_user_if_nonactive_long_enough(JID, TimeStampNow, SecOlder) ->
+    {LUser, LServer} = jid:to_lus(JID),
+    {ok, HostType} = mongoose_domain_api:get_domain_host_type(LServer),
+    case mod_last:get_last_info(HostType, LUser, LServer) of
+        {ok, TimeStamp, _Status} ->
+            %% get his age
+            Sec = TimeStampNow - TimeStamp,
+            %% If he is younger than SecOlder:
+            case Sec < SecOlder of
+                true ->
+                    %% do nothing
+                    false;
+                %% older:
+                false ->
+                    %% remove the user
+                    ejabberd_auth:remove_user(JID),
+                    true
+            end;
+        not_found ->
+            ejabberd_auth:remove_user(JID),
+            true
+    end.
+
+format_change_password(ok) ->
+    {ok, "Password changed"};
+format_change_password({error, empty_password}) ->
+    {empty_password, "Empty password"};
+format_change_password({error, not_allowed}) ->
+    {not_allowed, "Password change not allowed"};
+format_change_password({error, invalid_jid}) ->
+    {invalid_jid, "Invalid jid"}.
+
+-spec kick_sessions(jid:jid(), binary()) -> [ok].
+kick_sessions(JID, Reason) ->
+    lists:map(
+        fun(Resource) ->
+                service_admin_extra_sessions:kick_session(
+                  jid:replace_resource(JID, Resource), Reason)
+        end,
+        ejabberd_sm:get_user_resources(JID)).
+
+
+-spec set_random_password(JID, Reason) -> Result when
+      JID :: jid:jid(),
+      Reason :: binary(),
+      Result :: 'ok' | {error, any()}.
+set_random_password(JID, Reason) ->
+    NewPass = build_random_password(Reason),
+    ejabberd_auth:set_password(JID, NewPass).
+
+-spec build_random_password(Reason :: binary()) -> binary().
+build_random_password(Reason) ->
+    {{Year, Month, Day}, {Hour, Minute, Second}} = calendar:universal_time(),
+    Date = list_to_binary(
+             lists:flatten(
+               io_lib:format("~4..0w-~2..0w-~2..0wT~2..0w:~2..0w:~2..0wZ",
+                             [Year, Month, Day, Hour, Minute, Second]))),
+    RandomString = mongoose_bin:gen_from_crypto(),
+    <<"BANNED_ACCOUNT--", Date/binary, "--", RandomString/binary, "--", Reason/binary>>.
```

Comment:
That is kinda evil way to ban users, lol. Haaack.
That should be documented somewhere.

Reply:
It would be good to document it, but I think this falls out of scope of this PR as the code is unchanged here.